engine/access/rpc/backend/connection_factory.go

```diff
@@ -112,12 +165,17 @@ func (cf *ConnectionFactoryImpl) GetAccessAPIClient(address string) (access.Acce
 	return accessAPIClient, nil
 }
 
-func (cf *ConnectionFactoryImpl) InvalidateAccessAPIClient(address string) bool {
-	grpcAddress, err := getGRPCAddress(address, cf.CollectionGRPCPort)
-	if err != nil {
-		return true
+func (cf *ConnectionFactoryImpl) RefreshAccessAPIClient(address string) {
+	grpcAddress, _ := getGRPCAddress(address, cf.CollectionGRPCPort)
+	if res, ok := cf.ConnectionsCache.Get(grpcAddress); ok {
+		store := res.(ConnectionCacheStore)
+		store.mutex.Lock()
+		cf.lock.Lock()
+		defer store.mutex.Unlock()
+		defer cf.lock.Unlock()
+		store.ClientConn.Close()
+		_, _ = cf.addConnection(grpcAddress, cf.CollectionNodeGRPCTimeout, store.mutex)
```

Comment:
instead of adding it here, could we just close the connection and set it to nil? then the next time it's retrieved, we add the connection. This reduces the time spent with the lock
```
store := res.(ConnectionCacheStore)

store.mutex.Lock()

if store.conn != nil {
	conn := store.ClientConn
	store.ClientConn = nil
	defer conn.Close()
}

store.mutex.Unlock()
```